src/adaptors/amqp/qd_connector.c

```diff
@@ -439,27 +500,90 @@ void qd_connector_remove_connection(qd_connector_t *connector, bool final, const
+               "Connector %s new oldest valid TLS ordinal: %"PRIu64", previous: %"PRIu64,
                ctor_config->config.name, new_oldest_ordinal, ctor_config->tls_oldest_valid_ordinal);
+
+        // Spin through all connectors and delete those with an expired ordinal
+
+        qd_connector_t *ctor = DEQ_HEAD(ctor_config->connectors);
+        while (ctor) {
+            if (ctor->tls_ordinal < new_oldest_ordinal) {
+                qd_connector_t *delete_me = ctor;
+                ctor = DEQ_NEXT(ctor);
+                DEQ_REMOVE(ctor_config->connectors, delete_me);
+                qd_connector_close(delete_me);
+                qd_connector_decref(delete_me);
+            } else {
+                ctor = DEQ_NEXT(ctor);
+            }
+        }
+
         ctor_config->tls_oldest_valid_ordinal = new_oldest_ordinal;
     }
+
+    if (new_ordinal > ctor_config->tls_ordinal) {
+
+        qd_log(LOG_SERVER, QD_LOG_DEBUG,
+               "Connector %s new TLS ordinal: %"PRIu64", previous: %"PRIu64,
+               ctor_config->config.name, new_ordinal, ctor_config->tls_ordinal);
+
+        // Prevent existing connections that use older TLS ordinals from reconnecting. The goal is to allow older (but
+        // still valid) TLS sessions to pass existing flows but avoid sending new flows over them.
+
+        qd_connector_t *ctor = DEQ_HEAD(ctor_config->connectors);
+        while (ctor) {
+            bool active;
+            qd_connector_disable_reconnect(ctor, &active);
+            if (!active) {
+                // If the connector does not have an active connection it can be deleted
+                qd_connector_t *delete_me = ctor;
+                ctor = DEQ_NEXT(ctor);
+                DEQ_REMOVE(ctor_config->connectors, delete_me);
+                qd_connector_close(delete_me);
+                qd_connector_decref(delete_me);
+            } else {
+                ctor = DEQ_NEXT(ctor);
+            }
+        }
+
+        // Create a new set of connectors that will use the updated TLS credentials
+
+        ctor_config->tls_ordinal = new_ordinal;
+
+        for (int i = 0; i < ctor_config->data_connection_count; i++) {
+            ctor = qd_connector_create(ctor_config, true);
+            if (!ctor) {
+                // TODO: this should not happen in real life but if it does it cannot be handled gracefully. Complain
+                // loudly.
+                qd_log(LOG_SERVER, QD_LOG_ERROR, "Failed to create data Connector %s: resource allocation failed", ctor_config->config.name);
+            } else {
+                DEQ_INSERT_HEAD(ctor_config->connectors, ctor);
+                qd_connector_activate(ctor);
+            }
+        }
+
+        ctor = qd_connector_create(ctor_config, false);
+        if (!ctor) {
+                qd_log(LOG_SERVER, QD_LOG_ERROR, "Failed to create data Connector %s: resource allocation failed", ctor_config->config.name);
```

Comment:
```suggestion
                qd_log(LOG_SERVER, QD_LOG_ERROR, "Failed to create admin connector %s: resource allocation failed", ctor_config->config.name);
```

Reply:
We should leave out admin and just say "connector" becase the connector role may not be control (could be 'edge', etc).

Reply:
See latest patch